sdk/storage/azure-storage-common-data-movement/src/main/java/com/azure/storage/common/data/movement/PathScannerBuilder.java

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common.data.movement;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class PathScannerBuilder {
```

Comment:
- why do we need builder for a type as simple as PathScanner?
- is PathScanner going to be part of public contract or should it live in "implementation" package? if it's internal then builder isn't necessary.

Reply:
I paralleled the .NET revisions to change this to a Factory when implementing this into Java; @amnguye was there any particular reason we should keep it as a Factory?

If it's going to work like .NET, I'm guessing internal; and consequently, I should move it into implementation?

Reply:
Keep it as a factory in .NET? Since the scanner is for internal use for DMLib, it would be best to keep it as a factory for easier unit testing.

Reply:
imo, it's too early to say if factory is going to be useful or not for this until we know more about callers of this component. adding one shouldn't be hard, so maybe it should be part of the work where we attempt to use the scanner.